Make the mention a link to the exact comment. [legacy]

Zahuczky opened the pull request:
As the title says, the mention button, instead of just a plaintext @, makes the mention a link to the exact comment it's a reply to.

Zahuczky committed:
Make the mention a link to the exact comment.

Zahuczky commented:
>matey 
>Arrr! 
>It be the latest

What in tarnation is this summary

ArjixWasTaken commented:
I am kinda reluctant to merge this.
It is a great idea and your code doesn't look bad.

But this is the legacy code, but even then the legacy code is the one published to the extension stores...

This is a hard decision to make :(

ArjixWasTaken commented:
If you make additional PRs for the other two rewrites (look at the branches) I will merge this w/o hesitation!

Zahuczky commented:
I did take a look at your rewrite branches, but seeing they didn't progress for over a year, I figured this is the place to pr this, as this is the one in use by people.

You can also just keep this PR open, so people can discover this small change on their own and decide whether they want it or not.

ArjixWasTaken commented:
Also, after you do that, I'd like you to tell me which version you prefer, legacy, rewrite or rewrite2?

Zahuczky commented:
> If you make additional PRs for the other two rewrites (look at the branches) I will merge this w/o hesitation!

I can add this change to the rewrites too.

ArjixWasTaken commented:
I might have not finished the reply functionality in rewrite-v2 😃

Zahuczky commented:
As you've asked, I added it to the rewrite branch too, and yeah, the function in the rewrite-v2 was unfinished, but I finished it now.
I added a shorter description to how it works to the v2 PR, I hope this is how you've imagined this functionality.

Zahuczky commented:
Also,
> which version you prefer

In the rewrites, I didn't like that the options for the addons were in the extension settings, it was nicer when it was in nyaa. (tho this could be temporary I guess? not sure, wanted to mention)

In rewrite-"v1" I'm not sure the subscribe button should be there on every torrent in the homepage. Like, without even looking much at it, why would I want to subscribe to it?
Tho I guess you can argue that the popup gives you the option to take a look before that.

Regarding the popup, it's pretty cool, but to my taste might be a bit too obtrusive. Maybe it should be a button, or instead of holding your mouse over the title of the torrent, holding it over an icon or something. 

I liked the markdown editor in rewrite-v2. That's definitely very welcome.

Other than these, there's not much I noticed in the short while I tested the changes.

ArjixWasTaken commented:
> In the rewrites, I didn't like that the options for the addons were in the extension settings, it was nicer when it was in nyaa. (tho this could be temporary I guess? not sure, wanted to mention)

Not really temporary per-se, but more like, if smth goes wrong you can still access the extension settings.
I planned to also have the settings in the same place as the legacy version!

ArjixWasTaken commented:
> In rewrite-"v1" I'm not sure the subscribe button should be there on every torrent in the homepage. Like, without even looking much at it, why would I want to subscribe to it?
> Tho I guess you can argue that the popup gives you the option to take a look before that.
> 
> Regarding the popup, it's pretty cool, but to my taste might be a bit too obtrusive. Maybe it should be a button, or instead of holding your mouse over the title of the torrent, holding it over an icon or something.
> 
> I liked the markdown editor in rewrite-v2. That's definitely very welcome.
> 
> Other than these, there's not much I noticed in the short while I tested the changes.

I have two rewrites because I was not satisfied with the first rewrite.
The reason I kept both available is because I was going to develop both to some extent and pick one of them as the actual version.

It's been so long since I've worked on them, in your opinion which rewrite should be the main version?

Zahuczky commented:
Looking at your github it's clear that you're a much more experienced developer than I am, so I feel a bit out of place saying this, but are you sure all of these rewrites are really necessary?

I'd recommend reading this, as this has saved me from multiple complete rewrites whenever I wanted to change something non-trivial in a project: https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/

Also, given that I'm not a frontend dev myself, I might not be the best to ask which should eventually become the main version, but here's my 2 cents:
Legacy is clearly the most polished one, but given that you want to rewrite it no matter what, let's ignore that.

(Sidenote: I might be a bit partial to rewrite 2, as adding my thingy to the legacy and first rewrite codebase took a combined 10 minutes, I've spent hours forcing my way through the code of rewrite two[mainly the part to remove the mention{why the fuck it can't just access a basic function form the same class wtf(I don't like frontend)}] to implement it nicely)

Rewrite 1 seems to be more complete than rewrite 2, as that still lacks a lot of functionality.

I've only had a close look the in the reply source file of each, but quickly skimming through the other ones, I'm not really sure what's the "real" difference.

Yeah sure, it's different code, but overall they seem rather similar. 

Rewrite 2 uses Vite, which is cool last time I've heard? Not sure what actual benefits that brings.

Rewrite 2 seems to have more css tweaking, which is pretty cool.

At some places, rewrite 1 seems more straightforward.
I mean, it's less frontend webdev magic, and more "ah yeah thats code" which anyone can understand, without knowing JS/TS
(at least it looks like that to me)

I thought that after putting these in writing I could say what's my preference, but really, I'm not sure.

I like that legacy is already a working project.
I like that rewrite 1 does have most of that functionality already, in a bit nicer codebase.
I like that rewrite 2 has more and nicer UI tweaks.

But realistically, any one of these is a good candidate for further development.
